frontend/src/components/maps/leaflet/Map.tsx

```diff
@@ -215,8 +234,19 @@ const Map: React.FC<MapProps> = ({
   }, [interactive, mapRef]);
 
   const handleMapFilterChange = async (filter: IPropertyFilter) => {
-    setGeoFilter(getQueryParams(filter));
-    setChanged(true);
+    const compareValues = (objValue: any, othValue: any) => {
+      return whitelistedFilterKeys.reduce((acc, key) => {
+        return (
+          (_.isEqual(objValue[key], othValue[key]) ||
```

Comment:
this is a nitpick, but you import isEqual specifically from lodash but end up just using _ in the end.